fastserde/avro-fastserde/src/main/java/org/apache/avro/generic/ColdGenericDatumReader.java

```diff
@@ -10,10 +12,23 @@
  * This class needs to be in the org.apache.avro.generic package in order to access protected methods.
  */
 public class ColdGenericDatumReader<T> extends GenericDatumReader<T> implements ColdDatumReaderMixIn {
-  public ColdGenericDatumReader(Schema writerSchema, Schema readerSchema) {
+
+  private ColdGenericDatumReader(Schema writerSchema, Schema readerSchema, GenericData modelData) {
+    super(writerSchema, readerSchema, modelData != null ? modelData : GenericData.get());
+  }
+
+  private ColdGenericDatumReader(Schema writerSchema, Schema readerSchema) {
     super(writerSchema, readerSchema);
   }
 
+  public static <T> ColdGenericDatumReader<T> of(Schema writerSchema, Schema readerSchema, GenericData modelData) {
```

Comment:
I think that we are breaking the cohesion of interface for the *Reader classes by making the constructors private and by introducing the of() method.

Reply:
We must remember that *Reader and *Writer classes are part of the public interface of the library. We must refrain from introducing disruptive changes to them.

Reply:
Commit `[review-fixes] Part 5` restores public visibility of the constructors.

The static method `of()` could be placed somewhere else but I think it's quite natural to have it in `ColdGenericDatumReader` class.